llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -6905,6 +6905,17 @@ static bool planContainsAdditionalSimplifications(VPlan &Plan,
       if (isa<VPPartialReductionRecipe>(&R))
         return true;
 
+      // The VPlan-based cost model can analysis if recipes is scalar
+      // recursively, but legacy cost model cannot.
+      if (auto *WidenMemR = dyn_cast<VPWidenMemoryRecipe>(&R)) {
+        if (WidenMemR &&
+            vputils::isSingleScalar(WidenMemR->getAddr()) !=
+                CostCtx.CM.Legal->isUniform(
```

Comment:
Can this use `isLegacyUniformAfterVectorization`, as the code does below?
```suggestion
                CostCtx.isLegacyUniformAfterVectorization(
```